tests/console/libssh.pm

```diff
@@ -0,0 +1,151 @@
+#  * Run below commands to connect to libvirtd deamon.
+#      virsh -c "qemu+libssh://"
+#      virsh -c "qemu+libssh2://"
+#  * It's difficult to clean up libvirt so set test flag to rollback always
+#
+# Maintainer: Tony Yuan <tyuan@suse.com>
+
+package libssh;
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use Utils::Systemd 'disable_and_stop_service';
+use version_utils;
+use registration 'add_suseconnect_product';
+
+# Build a custom container image with openssl, curl and virsh installed.
+# The container will be used as client of libssh
+# The variable: dev_image_tag can be used to specify a base image different from host OS. If the base images is
+# stored in internel repository i.e registry.suse.de we assume the host OS is in develping stage i.e currently sle15sp3:
+# registry.suse.de/suse/sle-15-sp3/update/cr/totest/images/suse/sle15:15.3
+#
+sub create_image {
+    my $tag = get_var("DEV_IMAGE_TAG");
+    my $kubic;
+    if ($tag =~ /registry\.suse\.de/) {
+        assert_script_run("curl http://ca.suse.de/certificates/ca/SUSE_Trust_Root.crt -o /etc/pki/trust/anchors/SUSE_Trust_Root.crt");
+        assert_script_run("update-ca-certificates");
+        systemctl("restart docker.service");
+        $kubic = "zypper rm -y kubic-locale-archive";
+    }
+    my $dockerfile;
+    my $pkgs = "openssh curl libvirt-client";
+
+    if (is_sle) {
+        unless ($tag) {
+            $tag = "registry.suse.com/suse/";
+            my ($v, $sp) = get_os_release;
+            $tag .= is_sle('<15') ? "sles${v}sp$sp" : "sle${v}:${v}.$sp";
+        }
+        record_info($tag);
+
+        #Create Dockerfile for sle and opensuse
+        $dockerfile = <<'EOT';
+ARG tag
+FROM $tag
+ARG pkgs
+ARG kubic
+RUN zypper --gpg-auto-import-keys ref -s
+RUN $kubic
+RUN zypper install -y $pkgs
+RUN echo "root:test" | chpasswd
+RUN /usr/sbin/sshd-gen-keys-start
+CMD ["/usr/sbin/sshd", "-D"]
+EOT
+    } else {
+        $tag ||= (is_tumbleweed) ? "tumbleweed" : "leap";
+        $dockerfile = <<'EOT';
+ARG tag
+FROM opensuse/$tag
+ARG pkgs
+RUN zypper install -y $pkgs
+RUN echo "root:test" | chpasswd
+RUN /usr/sbin/sshd-gen-keys-start
+CMD ["/usr/sbin/sshd", "-D"]
+EOT
+    }
+    #Build an custom image with openssh curl libvirt-client sshpass
+    assert_script_run("mkdir /tmp/build && cd /tmp/build");
+    assert_script_run("cat > /tmp/build/Dockerfile <<'END'\n$dockerfile\nEND\n( exit \$?)");
+    assert_script_run(qq(docker build -t libssh_image --build-arg kubic="$kubic" --build-arg tag="$tag" --build-arg pkgs="$pkgs" .), timeout => 600);
+}
+
+sub run {
+    select_console 'root-console';
+    unless (is_opensuse) {
+        is_sle('<15') ? add_suseconnect_product("sle-module-containers", 12) : add_suseconnect_product("sle-module-containers");
+    }
+    # Host is used as server of libssh test
+    zypper_call("--gpg-auto-import-keys in docker libvirt-daemon-qemu qemu-kvm qemu-block-ssh");
```

Comment:
You can also simplify this test module by using `containers::common::install_docker_when_needed` a lot.

Reply:
Will use install_docker_when_needed but it seems that it has a bug.  The container extension uri for suseconnect is always 12 (sle-module-containers/12/x86_64). There are not 12.1, 12.2 ..... 
Line 88:  https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/containers/common.pm
add_suseconnect_product("sle-module-containers") will fail to add module. 
add_suseconnect_product("sle-module-containers", 12) should work.